### ISSUE TITLE
terminate called after throwing an instance of 'dmlc::ParamError'

### ISSUE THREAD

**Issue by apoyoman** [dmlc/xgboost#956]
Hi, after installing xgboost on Ubuntu (git clone --recursive https://github.com/dmlc/xgboost
cd xgboost; make -j4) and successfully running several times I obtained an error when I tried to start from a previous model:

xgboost machine3.conf model_in=xgb_tree_mc3_eta005_1.model model_out=xgb_tree_mc3_eta005_2.model > machine3-2.out 2>&1 &

The log file content is: 
[12:05:05] 74067x61 matrix with 1408801 entries loaded from train.crs
[12:05:05] boosting round 0, 1.19209e-06 sec elapsed
terminate called after throwing an instance of 'dmlc::ParamError'
  what():  Required parameter num_feature of int is not presented

The conf file content is:
# General Parameters, see comment for each definition
# choose the tree booster, can also change to gblinear
booster = gbtree
# this is the only difference with classification, use reg:linear to do linear classification
# when labels are in [0,1] we can also use reg:logistic
objective = reg:linear

# Tree Booster Parameters
# step size shrinkage
eta = 0.05 
# minimum loss reduction required to make a further partition
gamma = 1.0 
# minimum sum of instance weight(hessian) needed in a child
min_child_weight = 1 
# maximum depth of a tree
max_depth = 6 

# Task parameters
# the number of round to do boosting
num_round = 100
# 0 means do not save any model except the final round model
save_period = 50 
# The path of training data
data = "train.crs" 
# The path of validation data, used to monitor training process, here [test] sets name of the validation set
#eval[test] = "machine.txt.test" 
# evaluate on training data as well each round
eval_train = 1
# The path of test data 
test:data = "test.crs"      
eval_metric = "rmse"
nthread = 2

**Comment by tqchen** [comment]
Thanks for the catch, should be fixed by https://github.com/dmlc/xgboost/pull/959

**Comment by apoyoman** [comment]
Thanks! ... utilizing the same configuration file, etc, now xgboost takes about 4min per round as opposed to a second on my machine. (and no improvement of the score). I find this very strange since all the rounds up to this point of reloading the old model were very fast with steady improvement. Also, there is no evidence of pruning when trying to run with the old model. Here is the output of the log file for rounds 0,1,2:

[19:12:34] 74067x61 matrix with 1408801 entries loaded from train.crs
[19:12:34] boosting round 0, 1.66893e-06 sec elapsed
[19:16:32] [0]	train-rmse:0.464787
[19:16:32] boosting round 1, 238.048 sec elapsed
[19:19:43] [1]	train-rmse:0.464787
[19:19:43] boosting round 2, 428.552 sec elapsed

**Comment by tqchen** [comment]
OK, this should be fixed by this PR https://github.com/dmlc/xgboost/pull/962
Sorry about the trouble  and thanks for being patient. 

The slow speed is due to xgboost not caching predictions in the model, so first iteration could be slower.

**Comment by apoyoman** [comment]
Oh, oh, I think it must be some other problem (please let me know if you need any other information):

[09:54:39] 74067x61 matrix with 1408801 entries loaded from train.crs
[09:54:39] boosting round 0, 1.66893e-06 sec elapsed
[09:58:24] [0]	train-rmse:0.464787
[09:58:24] boosting round 1, 225.251 sec elapsed
[10:01:46] [1]	train-rmse:0.464787
[10:01:46] boosting round 2, 427.435 sec elapsed
[10:05:22] [2]	train-rmse:0.464787
[10:05:22] boosting round 3, 643.278 sec elapsed
[10:08:32] [3]	train-rmse:0.464787
[10:08:32] boosting round 4, 832.647 sec elapsed
[10:11:48] [4]	train-rmse:0.464787
[10:11:48] boosting round 5, 1029.19 sec elapsed

**Comment by tqchen** [comment]
oh really, Can you see if you can reproduce the same behavior on the toy dataset we provided? Say train for one iter, and continue train on that

**Comment by apoyoman** [comment]
[11:13:27] 6513x127 matrix with 143286 entries loaded from agaricus.txt.train
[11:13:27] 1611x127 matrix with 35442 entries loaded from agaricus.txt.test
[11:13:27] boosting round 0, 7.15256e-07 sec elapsed
[11:13:47] [0]	test-error:0.000000	train-error:0.001228
[11:13:47] boosting round 1, 20.4957 sec elapsed
[11:14:08] [1]	test-error:0.000000	train-error:0.001228
[11:14:08] update end, 40.8043 sec in all
^C[1]+  Done                    xgboost mushroom.conf model_in=models/0002.model > m1_cont.out 2>&1

**Comment by apoyoman** [comment]
This is exhibiting the same kind of behaviour: long time between rounds and no improvement on each round.

**Comment by apoyoman** [comment]
Did that information help? Is there anything else you would like me to do? (I tried the cli interface to xgboost because the docs say that one can continue training a previously trained model. The R interface does not have this).
